pytket/binders/pauli.cpp

```diff
@@ -282,42 +286,51 @@ PYBIND11_MODULE(pauli, m) {
           ":py:class:`Qubit` to :py:class:`Pauli`.",
           py::arg("map"), py::arg("coeff") = 1.)
       .def(
-          py::init<QubitPauliString, Complex>(),
+          py::init([](const SpPauliString &qps, const Complex &c) {
+            return SpCxPauliTensor(qps.string, c);
+          }),
           "Construct a QubitPauliTensor from a QubitPauliString.",
           py::arg("string"), py::arg("coeff") = 1.)
       .def(
           "__hash__",
-          [](const QubitPauliTensor &qps) { return hash_value(qps); })
-      .def("__repr__", &QubitPauliTensor::to_str)
-      .def("__eq__", &py_equals<QubitPauliTensor>)
-      .def("__ne__", &py_not_equals<QubitPauliTensor>)
-      .def("__lt__", &QubitPauliTensor::operator<)
-      .def(
-          "__getitem__", [](const QubitPauliTensor &qpt,
-                            const Qubit &q) { return qpt.string.get(q); })
-      .def(
-          "__setitem__", [](QubitPauliTensor &qpt, const Qubit &q,
-                            Pauli p) { return qpt.string.set(q, p); })
+          [](const SpCxPauliTensor &qps) { return qps.hash_value(); })
+      .def("__repr__", &SpCxPauliTensor::to_str)
+      .def("__eq__", &py_equals<SpCxPauliTensor>)
+      .def("__ne__", &py_not_equals<SpCxPauliTensor>)
+      .def("__lt__", &SpCxPauliTensor::operator<)
+      .def("__getitem__", &SpCxPauliTensor::get<QubitPauliMap>)
+      .def("__setitem__", &SpCxPauliTensor::set<QubitPauliMap>)
       .def(py::self * py::self)
-      .def(Complex() * py::self)
-      .def_readwrite(
-          "string", &QubitPauliTensor::string,
+      .def(
+          "__mul__",
```

Comment:
The old code `.def(Complex() * py::self)` caused this to be defined as `__rmul__`. I'm not sure if it matters but maybe best to use `__rmul__` here as well